metrics/network/network-metrics-iperf3.sh

```diff
@@ -0,0 +1,591 @@
+	local server_address=$(start_server "$image" "$server_command" "$server_extra_args")
+
+	# Verify server IP address
+	if [ -z "$server_address" ];then
+		clean_env
+		die "server: ip address no found"
+	fi
+
+	# Verify the iperf3 server is up
+	check_iperf3_server 1>&2
+
+	# and start the client container
+	local client_command="$init_cmds && iperf3 -J -u -c ${server_address} -l 64 -b 0 ${cli_args} -t ${transmit_timeout}"
+	local output=$(start_client "$image" "$client_command" "$client_extra_args")
+
+	clean_env 1>&2
+	echo "$output"
+}
+
+# Run a UDP PPS test between the host and a container, with the client on the host.
+# Use the smallest packets we can and run with unlimited bandwidth
+# mode to try and get as many packets through as possible.
+function get_host_cnt_pps() {
+	local cli_args="$1"
+
+	# Checks iperf3 tool installed in host
+	# We also need the json query tool, as we use JSON format results from iperf3
+	local cmds=("iperf3" "jq")
+	check_cmds "${cmds[@]}" 1>&2
+
+	# Initialize/clean environment
+	init_env 1>&2
+
+	# Make port forwarding
+	local server_extra_args="$server_extra_args -p $fwd_port"
+	local server_address=$(start_server "$image" "$server_command" "$server_extra_args")
+
+	# Verify server IP address
+	if [ -z "$server_address" ];then
+		clean_env
+		die "server: ip address no found"
+	fi
+
+	# Verify the iperf3 server is up
+	check_iperf3_server 1>&2
+
+	# and start the client container
+	local output=$(iperf3 -J -u -c $server_address -l 64 -b 0 -t $transmit_timeout "$cli_args")
+
+	clean_env 1>&2
+	echo "$output"
+}
+
+# This test measures the bandwidth between a container and the host.
+# where the container take the server role and the iperf3 client lives
+# in the host.
+function iperf3_host_cnt_bwd() {
+	local TEST_NAME="network bwd host contr"
+	local result="$(get_host_cnt_bwd)"
+	parse_iperf3_bwd "$TEST_NAME" "$result"
+}
+
+# This test is similar to "iperf3_host_cnt_bwd", the difference is this
+# tests runs in reverse mode.
+function iperf3_host_cnt_bwd_rev() {
+	local TEST_NAME="network bwd host contr reverse"
+	local result="$(get_host_cnt_bwd "-R")"
+	parse_iperf3_bwd "$TEST_NAME" "$result"
+}
+
+# This tests measures the bandwidth using different number of parallel
+# client streams. (2, 4, 8)
+function iperf3_multiqueue() {
+	local TEST_NAME="network multiqueue"
+	local client_streams=("2" "4" "8")
+
+	for s in "${client_streams[@]}"; do
+		tn="$TEST_NAME $s"
+		result="$(get_host_cnt_bwd "-P $s")"
+		sum="$(echo "$result" | grep "SUM" | tail -n2)"
+		parse_iperf3_bwd "$tn" "$sum"
+	done
+}
+
+# This test measures the packet-per-second (PPS) between two containers.
+# It uses the smallest (64byte) UDP packet streamed with unlimited bandwidth
+# to obtain the result.
+function iperf3_cnt_cnt_pps() {
+	local TEST_NAME="network pps cnt cnt"
+	local result="$(get_cnt_cnt_pps)"
+	parse_iperf3_pps "$TEST_NAME" "$result"
+	raw_iperf3_pps_results "$TEST_NAME" "$result"
```

Comment:
Oh, OK - I meant to add the raw results into the same .json results file as a new sub-section called 'raw', not into a separate file... so we would keep all the results from a single run together in the one results file.
Well, this will do for now - I'll open an Issue to note that all iperf3 results should probably run in -J json mode and store their raw data in their single results file.